encoder/v1/validator.go

```diff
@@ -0,0 +1,54 @@
+package v1
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+)
+
+type validator struct{}
+
+// ValidateUpkeepKey returns true if the types.UpkeepKey is valid, false otherwise
+func (v validator) ValidateUpkeepKey(key types.UpkeepKey) (bool, error) {
+	blockKey, upkeepIdentifier, err := key.BlockKeyAndUpkeepID()
+	if err != nil {
+		return false, err
+	}
+
+	_, err = v.ValidateBlockKey(blockKey)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = v.ValidateUpkeepIdentifier(upkeepIdentifier)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// ValidateUpkeepIdentifier returns true if the types.UpkeepIdentifier is valid, false otherwise
+func (v validator) ValidateUpkeepIdentifier(identifier types.UpkeepIdentifier) (bool, error) {
```

Comment:
should we have a check that it's not greater than uint256.max here too? (not fully sure but i think we had that before?)

Reply:
we have this in main https://github.com/smartcontractkit/ocr2keepers/blob/main/pkg/chain/upkeep_observation.go#L41-L57 not sure if this check is still present somewhere else